website/api/controllers/try-fleet/view-query-report.js

```diff
@@ -0,0 +1,237 @@
+module.exports = {
+
+
+  friendlyName: 'View query report',
+
+
+  description: 'Display "Query report" page.',
+
+  inputs: {
+
+    hostPlatform: {
+      type: 'string',
+      required: true,
+      description: 'The platform of the host to display results for',
+      extendedDescription: '',
+      isIn: ['macos', 'linux', 'windows']
+    },
+
+    tableName: {
+      type: 'string',
+      required: true,
+      description: 'The name of the osquery table to show results for.',
+    },
+
+  },
+
+
+  exits: {
+
+    success: {
+      viewTemplatePath: 'pages/try-fleet/query-report'
+    },
+
+    badConfig: {
+      responseType: 'badConfig'
+    },
+
+    redirect: {
+      description: 'The requesting user is not logged in.',
+      responseType: 'redirect'
+    },
+
+    invalidTable: {
+      responseType: 'notFound',
+      description: 'No osquery table with the specified name could be found.'
+    },
+
+  },
+
+
+  fn: async function ({hostPlatform, tableName}) {
+
+    if(!_.isObject(sails.config.builtStaticContent) || !_.isArray(sails.config.builtStaticContent.osqueryTables)){
+      throw {badConfig: 'builtStaticContent.osqueryTables'};
+    }
+
+    // If the requesting user is not logged in, redirect them to the /try-fleet/login page with the specified hostPlatform added as a query parameter.
+    if(!this.req.me){
+      throw {redirect: `/try-fleet/login?targetPlatform=${hostPlatform}` };
```

Comment:
warning danger danger

Reply:
technically the isIn saves you

Reply:
either way, the solution is to use `encodeURIComponent()`.  Even though you don't technically need it here, it's just good hygiene (and prevents needless worry- also in the future the code could change and the worry could become needful)